libevmasm/Assembly.cpp

```diff
@@ -74,6 +74,117 @@ unsigned Assembly::codeSize(unsigned subTagSize) const
 	}
 }
 
+void Assembly::addAssemblyItemsFromJSON(Json::Value const& _code)
+{
+	solAssert(_code.isArray(), "");
+	for (auto const& jsonItem: _code)
+		m_items.emplace_back(loadItemFromJSON(jsonItem));
+
+	for (auto current = m_items.begin(); current != m_items.end(); ++current)
+	{
+		// During the assembly json export a `JUMPDEST` is always generated after a `tag`.
+		// So we just ignore exactly these `JUMPDEST`'s.
+		auto const next = std::next(current);
+		if (
+			next != m_items.end() &&
+			current->type() == AssemblyItemType::Tag &&
+			next->type() == AssemblyItemType::Operation &&
+			next->instruction() == Instruction::JUMPDEST
+		)
+			m_items.erase(next);
+	}
+}
+
+AssemblyItem Assembly::loadItemFromJSON(Json::Value const& _json)
+{
+	std::string name = _json["name"].isString() ? _json["name"].asString() : "";
```

Comment:
Shouldn't these all fail if the type is wrong? Unless it is optional, of course.

Reply:
Also could use some helpers `asStringOrDefault` / `asIntOrDefault`?

Reply:
Yeah those helpers would make sense!

Reply:
True, only the 'name' is asserted so far. Need add an `assert` for the other members.

Reply:
I introduced in `libsolutil/JSON.h` the helper functions `ofType`, `ofTypeIfExists` and `getOrDefault`. These functions are now used here.